src/Session/SessionManager.cc

```diff
@@ -88,7 +88,7 @@ void SessionManager::OnConnect(WSType* ws) {
     // create a Session
     std::unique_lock<std::mutex> ulock(_sessions_mutex);
     _sessions[session_id] = new Session(ws, loop, session_id, address, _settings.top_level_folder, _settings.starting_folder,
-        _file_list_handler, _settings.read_only_mode, _settings.enable_scripting);
+        _file_list_handler, _settings.read_only_mode, _settings.enable_scripting, _settings.controller_deployment);
```

Comment:
While this does work, and we do already pass some settings values into sessions like this, I would prefer not to change the signature of the session constructor for a property that is only used in telemetry. Please use the same method that we use to set the exit timeouts: make the variable *static*, and set it using a *static function* on the Session class. This is already being done with the two exit timeout properties in Main, after the settings are parsed (not per session in SessionManager).

(These other existing settings should probably also be removed from this constructor and set as static properties, but that is outside the scope of this PR -- I will file a separate code maintenance issue for that.)